File: matlab/measurement.m
function y = measurement(x)
    g = 9.81;
    theta = x(1);
    dtheta = x(2);
    dphi = x(4);

    y = zeros(2, 1);
    y(1) = g * sin(theta);
    y(2) = dtheta + dphi;
end

File: components/logic/measurement/measurement.c
#include "measurement.h"
#include <math.h>

void measurement_model(const state_t * x, const dynamics_params_t * params, measurement_t * y)
{
  float g, theta, dtheta, dphi;
  g = params->g;
  theta = x->theta;
  dtheta = x->dtheta;
  dphi = x->dphi;

  y->accel_x = g * sinf(theta);
  y->gyro_z = dtheta + dphi;
}